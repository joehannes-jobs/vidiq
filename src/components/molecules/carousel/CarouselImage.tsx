import React, { PropsWithChildren } from 'react';

export interface CarouselImageProps {
  albumId: number;
  id: number;
  title: string;
  thumbnailUrl: string;
}

const CarouselImage: React.FC<CarouselImageProps> = ({
  albumId,
  id: imgId,
  title,
  thumbnailUrl: url,
}: PropsWithChildren<CarouselImageProps>): JSX.Element => (
  <div key={`album-${albumId}-img-${imgId}`} className={`carousel-item`}>
    <img src={url} alt={title} />
  </div>
);

export default CarouselImage;
