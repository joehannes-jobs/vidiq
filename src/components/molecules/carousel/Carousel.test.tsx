import React from 'react';
import nock from 'nock';
import { shallow, mount } from 'enzyme';
import { waitFor } from '@testing-library/react';

import { Component as Carousel, API } from '.';
import CarouselImage from './CarouselImage';
import ImageData from './stub.json';

const ImageAPIBaseUrl = API.BASE_URL;

// increase timeout: the big .json test is taking some time
jest.setTimeout(21000);

describe('Carousel', () => {
  let fullyRenderedComponent: any;
  let shallowRenderedComponent: any;
  const scope: any = nock(ImageAPIBaseUrl)
    .defaultReplyHeaders({
      'access-control-allow-origin': '*',
      'access-control-allow-credentials': 'true',
    })
    .persist();

  beforeAll(() => {
    scope.persist().get('/photos').reply(200, ImageData.slice(0, 10), {
      'Content-Type': 'application/json',
    });

    shallowRenderedComponent = shallow(<Carousel />);
    fullyRenderedComponent = mount(<Carousel />);
  });

  afterAll(() => {
    scope.done();
  });

  it('should render the Carousel Component', () => {
    expect(fullyRenderedComponent.equals(<Carousel />)).toEqual(true);
  });

  it('should render a single toplevel <section /> tag', () => {
    expect(shallowRenderedComponent.find('section').length).toEqual(1);
  });

  it('should render initial Layout', () => {
    expect(shallowRenderedComponent.getElements()).toMatchSnapshot();
  });

  it('should load data and render <CarouselImage />', async () => {
    await waitFor(
      () => {
        fullyRenderedComponent.update();
        expect(
          fullyRenderedComponent.containsAnyMatchingElements(
            ImageData.slice(0, 10).map(dataset => (
              <CarouselImage
                key={`album-${dataset.albumId}-img-${dataset.id}`}
                title={dataset.title}
                thumbnailUrl={dataset.thumbnailUrl}
                onClick={jest.fn}
              />
            ))
          )
        ).toEqual(true);
      },
      { interval: 50, timeout: 3000 }
    );
  });
});
