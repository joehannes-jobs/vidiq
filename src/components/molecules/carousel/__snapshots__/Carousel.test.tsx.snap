// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render initial Layout 1`] = `
Array [
  <section
    className="w-auto h-screen
                  carousel carousel-center
                  portrait:carousel-vertical
                  landscape:md:carousel-vertical"
  >
    Loading images...
  </section>,
]
`;
